runtime/src/main/java/org/apache/flink/agents/runtime/operator/ActionExecutionOperator.java

```diff
@@ -301,20 +309,47 @@ private void processActionTaskForKey(Object key) throws Exception {
 
         // 2. Invoke the action task.
         createAndSetRunnerContext(actionTask);
-        ActionTask.ActionTaskResult actionTaskResult = actionTask.invoke();
-        for (Event actionOutputEvent : actionTaskResult.getOutputEvents()) {
+
+        boolean isFinished = false;
+        List<Event> outputEvents;
+        Optional<ActionTask> generatedActionTaskOpt;
+        ActionState actionState = actionStateStore.get(key, actionTask.action);
+        if (actionState != null) {
+            isFinished = actionState.getGeneratedActionTask().isPresent();
+            outputEvents = actionState.getOutputEvents();
+            generatedActionTaskOpt = actionState.getGeneratedActionTask();
+            for (MemoryUpdate memoryUpdate : actionState.getMemoryUpdates()) {
+                actionTask
+                        .getRunnerContext()
+                        .getShortTermMemory()
+                        .set(memoryUpdate.getPath(), memoryUpdate.getValue());
+            }
+        } else {
```

Comment:
In python async execution, if an action task is invoked but not finished, it will be put back to `actionTasksKState` and re-invoked later. In that case, it will falls into the `if (actionState != null)` clause, and treated as replay rather than resume.